torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -38,16 +40,36 @@ const DistAutogradContext& DistAutogradContainer::newContext() {
   }
 
   std::lock_guard<std::mutex> guard(autograd_context_lock_);
-  if (current_context_id_ == std::numeric_limits<int64_t>::max() ||
-      current_context_id_ >
+  if (next_context_id_ == std::numeric_limits<int64_t>::max() ||
+      next_context_id_ >
           (kMaxContextId |
            (static_cast<int64_t>(worker_id_) << kContextIdBits))) {
     throw std::runtime_error("We have run out of autograd context ids!!!");
```

Comment:
nit: Not part of this PR, but we could have used a `TORCH_INTERNAL_ASSERT` here

Reply:
(You don't have to do anything to this code, I'm just letting you know in case you need to write something similar in the future)